cmd/engine/main_test.go

```diff
@@ -43,3 +45,91 @@ func TestParallelismFlag(t *testing.T) {
 		require.Error(t, err)
 	})
 }
+
+func TestEngineNameLabel(t *testing.T) {
```

Comment:
This test is admittedly so "unit" that it's barely useful, I'd prefer an integ test but the only way to do that would be to use a custom provisioned engine in docker, so I think it's better to just wait for this to be merged: https://github.com/dagger/dagger/pull/4848